Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -1565,6 +1565,201 @@ def getResultSummary(self):
+                content_type='text/plain',
+            ), UploadFileToS3(
+                'build-log.txt',
+                links={self.name: 'Full build log'},
+                content_type='text/plain',
+            )
+        ]
+        self.build.addStepsAfterCurrentStep(steps_to_add)
+
+    def evaluateCommand(self, cmd):
+        if cmd.rc != 0:
+            self.commandFailed = True
+            return FAILURE
+        if self.analyzeFailed:
+            return FAILURE
+
+        return SUCCESS
+
+    def getResultSummary(self):
+        status = ''
+        if self.analyzeFailed:
+            status += 'ANALYZE FAILED: '
+        status += self.name
+
+        if self.commandFailed:
+            status += ' failed to build, and'
+        status += ' found {} issues'.format(self.bugs)
+
+        if self.results != SUCCESS:
+            status += ' ({})'.format(Results[self.results])
+        else:
+            self.build.addStepsAfterLastStep([UpdateSmartPointerBaseline()])
+
+        return {u'step': status}
+
+
+class ParseStaticAnalyzerResults(shell.ShellCommand):
+    name = 'parse-static-analyzer-results'
+    description = ['parsing static analyzer results']
+    descriptionDone = ['parsed static analyzer results']
+    resultMsg = ''
+
+    def start(self):
+        self.command = 'python3 Tools/Scripts/generate-dirty-files {scan_build_output} '.\
+            format(scan_build_output=os.path.join(self.getProperty('builddir'), 'build/scan-build-output'))
+        output_dir = 'smart-pointer-result-archive/{}'.format(self.getProperty('buildnumber'))
+        self.command += '--output-dir {output_dir} '.format(output_dir=os.path.join(self.getProperty('builddir'), output_dir))
+        self.command += '--build-dir {build_dir}'.format(build_dir=os.path.join(self.getProperty('builddir'), 'build'))
+
+        self.log_observer = logobserver.BufferLogObserver()
+        self.addLogObserver('stdio', self.log_observer)
+        return shell.ShellCommand.start(self)
+
+    def commandComplete(self, cmd):
+        shell.ShellCommand.commandComplete(self, cmd)
+
+        log_text = self.log_observer.getStdout()
+        index = log_text.rfind('Total')
+        self.resultMsg = log_text[index:]
+
+    def evaluateCommand(self, cmd):
+        if cmd.rc != 0:
+            self.commandFailed = True
+            return FAILURE
+        return SUCCESS
+
+    def getResultSummary(self):
+        status = ''
+        if self.resultMsg:
+            status += ' Issues by project: {}'.format(self.resultMsg)
+        if self.results != SUCCESS:
+            status += ' ({})'.format(Results[self.results])
+
+        return {u'step': status}
+
+
+class CompareStaticAnalyzerResults(shell.ShellCommand):
+    name = 'compare-static-analyzer-results'
+    description = ['comparing static analyzer results']
+    descriptionDone = ['compared static analyzer results']
+    resultMsg = ''
+
+    command = 'python3 Tools/Scripts/compare-static-analysis-results '
```

Comment:
It doesn't look like we're doing any piping, so this should be a list instead of a string.